lib/plugins/revhistory/_test/plugin_revhistory.test.php

```diff
@@ -0,0 +1,460 @@
+<?php
+/**
+ * Require the DokuWiki_Action_Plugin
+ */
+require_once(DOKU_INC . 'lib/plugins/revhistory/action.php');
+
+/**
+ * Tests to ensure the plugin works as advertised
+ * DokuWiki Plugin revhistory
+ *
+ * @license     Copyright (c) 2015 unfoldingWord http://creativecommons.org/licenses/MIT/
+ * @author      Johnathan Pulos <johnathan@missionaldigerati.org>
+ *
+ * @group plugin_revhistory
+ * @group door43_plugins
+ * @group plugins
+ */
+class plugin_revhistory_test extends DokuWikiTest
+{
+    /**
+     * Our plugin to test
+     *
+     * @var object
+     * @access protected
+     **/
+    protected $plugin;
+
+    /**
+     * A mock of the Doku_Event Object
+     *
+     * @var object
+     * @access protected
+     **/
+    protected $eventHandler;
+
+    /**
+     * Setup the testing environment
+     *
+     * @author Johnathan Pulos <johnathan@missionaldigerati.org>
+     */
+    public function setUp()
+    {
+        global $conf;
+        parent::setUp();
+        $this->plugin = new action_plugin_revhistory();
+        $this->plugin->testing = true;
+        $this->eventHandler = $this->getMockBuilder('Doku_Event')
+            ->setConstructorArgs(array('ACTION_ACT_PREPROCESS', 'revhistory'))
+            ->getMock();
+        /**
+         * Overwrite the config directories so it uses our test directories.
+         * For some reason when DokuWiki moves the test files, they do not get moved and throw an error.
+         */
+        $conf['changelog']          = dirname(__FILE__) . '/data/meta/_dokuwiki.changes';
+        $conf['media_changelog']    = dirname(__FILE__) . '/data/meta/_media.changes';
```

Comment:
Thank you Phillip.  I tried that, and when I try to open the config files it throws a file not found error.  I looked in the tmo folder, and the data folder isn't there.  According to the docs, it should move the files to temp folder automatically, without rcopy command,  but it doesn't work either.   I still get a file not found error.  Look at the Environment section in the testing doc: https://www.dokuwiki.org/devel:unittesting.  I am on a Mac.  Does it work for you?